pipeline/frontend/yaml/compiler/compiler.go

```diff
@@ -139,14 +141,6 @@ func (c *Compiler) Compile(conf *yaml_types.Workflow) (*backend_types.Config, er
 		Name: fmt.Sprintf("%s_default", c.prefix),
 	})
 
-	// create secrets for mask
-	for _, sec := range c.secrets {
-		config.Secrets = append(config.Secrets, &backend_types.Secret{
```

Comment:
this is finally gone :tada: so the agent only has to know the secrets that are actually used